packages/dashboard/src/lib/framework/extension-api/types/navigation.ts

```diff
@@ -4,16 +4,56 @@ import type React from 'react';
 
 import { NavMenuItem } from '../../nav-menu/nav-menu-extensions.js';
 
+/**
+ * @description
+ * Defines a custom route for the dashboard with optional navigation menu integration.
+ */
 export interface DashboardRouteDefinition {
+    /**
+     * @description
+     * The React component that will be rendered for this route.
+     */
     component: (route: AnyRoute) => React.ReactNode;
+    /**
+     * @description
+     * The URL path for this route, e.g. '/my-custom-page'.
+     */
     path: string;
+    /**
+     * @description
+     * Optional navigation menu item configuration to add this route to the sidebar.
+     */
     navMenuItem?: Partial<NavMenuItem> & { sectionId: string };
+    /**
+     * @description
+     * Optional loader function to fetch data before the route renders.
```

Comment:
You can note that it expect a route loader from Tanstack Router (https://tanstack.com/router/latest/docs/framework/react/guide/data-loading#route-loaders)